### ISSUE TITLE
Persistence of projections with composite properties

### ISSUE THREAD

**Issue by gonzalad** [spring-projects/spring-data-neo4j#2451]
Hello,

I'm trying to use persistence of projections on an entity defining a composite property, but the property is not saved.

I'm using SDN 6.2.0.

Here is my testcase:

```
@Node("Widget")
public class WidgetEntity {

    @GeneratedValue
    @Id
    private Long id;
    private String code;
    private String label;

    @CompositeProperty
    private Map<String, Object> additionalFields = new HashMap<>();

...
}
```  

```
public interface WidgetProjection {

    String getCode();

    String getLabel();

    Map<String, Object> getAdditionalFields();
}
```

Unit test:

```
    @Test
    void testSaveAsWidgetProjectionWithCompositeProperty() {

        String code = "Window1";
        WidgetEntity window = repository.findByCode(code).orElseThrow();
        window.setLabel("changed");
        window.getAdditionalFields().put("key1", "value1");

        template.saveAs(window, WidgetProjection.class);

        window = repository.findByCode(code).orElseThrow();
        assertThat(window.getAdditionalFields()).hasSize(1); => this returns 0 !!
    }
```

The code is available in this repository: https://github.com/gonzalad/sdn6-tests/blob/projection-persist-composite-properties/src/test/java/com/example/sdn6/Sdn6Test.java (it relies on testcontainers: no special setup for neo4j db required)

Thanks,

N.B. the issue seems to be related to TemplateSupport.createAndApplyPropertyFilter which gets additionalFields.key1 in the NAME_OF_PROPERTIES_PARAM parameter and tries to compare it with the composite property additionalFields

**Comment by meistermeier** [comment]
Thanks for reporting this.
I would link this to #2371 because as far as I understood what is happening, both issues are very close.

**Comment by surjitbhachu** [comment]
I've hit this problem as well, composite properties are not saving when using `saveAs`. Are there any workarounds to save these properties to the database?

**Comment by cberes** [comment]
Just encountered this problem. We use CompositeProperty with and without a converter, and both scenarios are affected.

**Comment by michael-simons** [comment]
Thanks everyone for reporting this. It will be fixed from 6.1 onwards.
